Issue #15449: excluded rule411optionalbracesusage from google-java-format.sh

Zopsss opened the pull request:
~~#15340~~ #15449 


~~the formatted file contains many new violations regarding indentation~~

excluded section 4.1.1 from google-java-format.sh

rdiachenko commented on src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputFormattedUseOfOptionalBraces.java:
```diff
@@ -0,0 +1,293 @@
+    while (condition())
+      ; // violation above ''while' construct must use '{}'s.'
+    while (condition()) // violation ''while' construct must use '{}'s.'
+    testWhile(); // violation ''while' child has incorrect indentation level 4, expected .* 6.'
```
So, checkstyle is right here and in other similar cases below

romani replied:
This can be resolved by formatter, I am not sure why they did not target it. 
Let's recheck with formatter by creating issue on them

romani replied:
We should treat violations on formatting after formatter as red frag on our misinterpretation of style.

Zopsss replied:
there are 2 bugs of formatter here:

1. No usage of `{}` ( https://github.com/google/google-java-format/issues/51 )
2. Incorrect indentation for 'while' child ( https://github.com/google/google-java-format/issues/1009, this is a similar issue to our problem )

The issues already seems to be reported to them

romani replied:
lets keep this discussion unresovled to ease finding reasons why there are violation later on.

rdiachenko reviewed (approved):
lgtm

romani commented on src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputFormattedUseOfOptionalBraces.java:
```diff
@@ -0,0 +1,293 @@
+
+    // Invalid
+    do testDoWhile(); // violation ''do' construct must use '{}'s.'
+    while (condition());
```
Is this formatter or checkstyle problem?
It might be that we misinterprets guide

Zopsss replied:
the missing `{}` is their problem and if you're talking about indentation of `while` then it is correct, `while` isn't child of `do` so the indentation should not be increased for it

romani replied:
good, thanks a lot.

romani reviewed (changes requested):
Items

romani reviewed (changes requested):
please fix in Original Input all indentation violations. they are not related.

![image](https://github.com/user-attachments/assets/770f6154-15d2-471d-92b0-878be52c0a1b)

it will simplify a diff.

Zopsss commented:
> please fix in Original Input all indentation violations. they are not related.

there are no indentation violations in original files 👀

![Screenshot 2024-08-31 222745](https://github.com/user-attachments/assets/46d2367c-3201-4107-8692-7ef121a0e14e)

romani commented:
If indentation is damaged my formatter, it is defect on it. So there in comments should be issue to it.

romani commented:
please update original Input file from:
```
    while (condition()) // violation ''while' construct must use '{}'s.'
      testWhile();
```

to 
```
   // violation below ''while' construct must use '{}'s.' 
   while (condition())
      testWhile();
```

to avoid comments inside code that we are violating, AND it will let formatter to operate easily.

Apply this to all cases in file.

Zopsss commented:
> Apply this to all cases in file.

done, changed violation messages in OG file and based on that created new formatted file, formatter changed location of some violations in formatted file


```diff
$ diff -u src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputFormattedUseOfOptionalBraces.java
--- src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputUseOfOptionalBraces.java     2024-09-01 22:56:57.286227800 +0530
+++ src/it/resources/com/google/checkstyle/test/chapter4formatting/rule411optionalbracesusage/InputFormattedUseOfOptionalBraces.java    2024-09-01 22:56:14.993383100 +0530
@@ -1,6 +1,6 @@
 package com.google.checkstyle.test.chapter4formatting.rule411optionalbracesusage;
 
-class InputUseOfOptionalBraces {
+class InputFormattedUseOfOptionalBraces {
   /**
    * some javadoc..
    *
@@ -32,15 +32,13 @@

     // Invalid
     // violation 2 lines below ''while' construct must use '{}'s.'
-    // violation 2 lines below ''while' construct must use '{}'s.'
-    while (condition());
+    // violation 3 lines below ''while' construct must use '{}'s.'
     while (condition())
-      testWhile();
+      ;
+    while (condition()) testWhile();
     // violation 2 lines below ''while' construct must use '{}'s.'
-    // violation 2 lines below ''if' construct must use '{}'s.'
-    while (condition())
-      if (condition())
-          testWhile();
+    // violation below ''if' construct must use '{}'s.'
+    while (condition()) if (condition()) testWhile();
     String k = "testing";
     // violation below ''if' construct must use '{}'s.'
     if (k != null) k = "ss";
@@ -55,17 +53,13 @@

     // Invalid
     // violation below ''for' construct must use '{}'s.'
-    for (int i = 1; i < 5; i++);
-    // violation below ''for' construct must use '{}'s.'
     for (int i = 1; i < 5; i++)
-      testFor();
+      ;
+    // violation below ''for' construct must use '{}'s.'
+    for (int i = 1; i < 5; i++) testFor();
     // violation 2 lines below ''for' construct must use '{}'s.'
-    // violation 4 lines below ''if' construct must use '{}'s.'
-    for (int i = 1;
-        i < 5;
-        i++)
-      if (i > 2)
-        testFor();
+    // violation below ''if' construct must use '{}'s.'
+    for (int i = 1; i < 5; i++) if (i > 2) testFor();
   }

   /** Test if constructs. */
@@ -81,32 +75,26 @@

     // Invalid
     // violation below ''if' construct must use '{}'s.'
-    if (condition());
-    // violation below ''if' construct must use '{}'s.'
     if (condition())
-      testIf();
+      ;
+    // violation below ''if' construct must use '{}'s.'
+    if (condition()) testIf();
     // violation 2 lines below ''if' construct must use '{}'s.'
-    // violation 3 lines below ''else' construct must use '{}'s.'
-    if (condition())
-      testIf();
-    else
-      testIf();
+    // violation 2 lines below ''else' construct must use '{}'s.'
+    if (condition()) testIf();
+    else testIf();
     // violation below ''if' construct must use '{}'s.'
-    if (condition())
-      testIf();
+    if (condition()) testIf();
     else {
       testIf();
     }
     if (condition()) {
       testIf();
       // violation below ''else' construct must use '{}'s.'
-    } else
-      testIf();
-    // violation 2 lines below ''if' construct must use '{}'s.'
+    } else testIf();
     // violation 2 lines below ''if' construct must use '{}'s.'
-    if (condition())
-      if (condition())
-          testIf();
+    // violation below ''if' construct must use '{}'s.'
+    if (condition()) if (condition()) testIf();
   }

   void whitespaceAfterSemi() {
@@ -119,7 +107,7 @@
   }

   /** Empty constructor block. */
-  public InputUseOfOptionalBraces() {}
+  public InputFormattedUseOfOptionalBraces() {}

   /** Empty method block. */
   public void emptyImplementation() {}
@@ -137,33 +125,40 @@
   class EmptyBlocks {
     boolean flag = true;
     int[] abc = {
-        1, 2, 3, 4,
+      1, 2, 3, 4,
     };

     void foo() {
       // violation below ''while' construct must use '{}'s.'
-      while (flag);
+      while (flag)
+        ;
       while (flag) {}
       while (flag) {
         /*foo*/
       }
       // violation below ''do' construct must use '{}'s.'
-      do ;
+      do
+        ;
       while (flag);
       do {} while (flag);
       do {
         /*foo*/
       } while (flag);
       // violation below ''if' construct must use '{}'s.'
-      if (flag) ;
-      if (flag) { /*foo*/ }
+      if (flag)
+        ;
+      if (flag) {
+        /*foo*/
+      }
       if (flag) {
         /*foo*/
       }
       // violation 2 lines below ''if' construct must use '{}'s.'
-      // violation 2 lines below ''else' construct must use '{}'s.'
-      if (flag) ;
-      else ;
+      // violation 3 lines below ''else' construct must use '{}'s.'
+      if (flag)
+        ;
+      else
+        ;
       // violation below 'Empty if block.'
       if (flag) {
       } else {
@@ -175,13 +170,15 @@
         /*foo*/
       }
       // violation below ''for' construct must use '{}'s.'
-      for (int i = 0; i < 10; i++);
+      for (int i = 0; i < 10; i++)
+        ;
       for (int i = 0; i < 10; i++) {}
       for (int i = 0; i < 10; i++) {
         /*foo*/
       }
       // violation below ''for' construct must use '{}'s.'
-      for (int b : abc);
+      for (int b : abc)
+        ;
       for (int b : abc) {}
       for (int b : abc) {
         /*foo*/
@@ -191,33 +188,40 @@
     class InnerEmptyBlocks {
       boolean flag = true;
       int[] abc = {
-          1, 2, 3, 4,
+        1, 2, 3, 4,
       };

       void foo() {
         // violation below ''while' construct must use '{}'s.'
-        while (flag);
+        while (flag)
+          ;
         while (flag) {}
         while (flag) {
           /*foo*/
         }
         // violation below ''do' construct must use '{}'s.'
-        do ;
+        do
+          ;
         while (flag);
         do {} while (flag);
         do {
           /*foo*/
         } while (flag);
         // violation below ''if' construct must use '{}'s.'
-        if (flag) ;
-        if (flag) { /*foo*/ }
+        if (flag)
+          ;
+        if (flag) {
+          /*foo*/
+        }
         if (flag) {
           /*foo*/
         }
         // violation 2 lines below ''if' construct must use '{}'s.'
-        // violation 2 lines below ''else' construct must use '{}'s.'
-        if (flag) ;
-        else ;
+        // violation 3 lines below ''else' construct must use '{}'s.'
+        if (flag)
+          ;
+        else
+          ;
         // violation below 'Empty if block.'
         if (flag) {
         } else {
@@ -229,13 +233,15 @@
           /*foo*/
         }
         // violation below ''for' construct must use '{}'s.'
-        for (int i = 0; i < 10; i++);
+        for (int i = 0; i < 10; i++)
+          ;
         for (int i = 0; i < 10; i++) {}
         for (int i = 0; i < 10; i++) {
           /*foo*/
         }
         // violation below ''for' construct must use '{}'s.'
-        for (int b : abc);
+        for (int b : abc)
+          ;
         for (int b : abc) {}
         for (int b : abc) {
           /*foo*/
@@ -244,59 +250,68 @@
     }

     InnerEmptyBlocks anon =
-            new InnerEmptyBlocks() {
-              boolean flag = true;
-              int[] abc = {
-                  1, 2, 3, 4,
-              };
-
-              void foo() {
-                // violation below ''while' construct must use '{}'s.'
-                while (flag);
-                while (flag) {}
-                while (flag) {
-                  /*foo*/
-                }
-                // violation below ''do' construct must use '{}'s.'
-                do ;
-                while (flag);
-                do {} while (flag);
-                do {
-                  /*foo*/
-                } while (flag);
-                // violation below ''if' construct must use '{}'s.'
-                if (flag) ;
-                if (flag) { /*foo*/ }
-                if (flag) {
-                  /*foo*/
-                }
-                // violation 2 lines below ''if' construct must use '{}'s.'
-                // violation 2 lines below ''else' construct must use '{}'s.'
-                if (flag) ;
-                else ;
-                // violation below 'Empty if block.'
-                if (flag) {
-                } else {
-                }
-                // violation 2 lines above 'Empty else block.'
-                if (flag) {
-                  /*foo*/
-                } else {
-                  /*foo*/
-                }
-                // violation below ''for' construct must use '{}'s.'
-                for (int i = 0; i < 10; i++);
-                for (int i = 0; i < 10; i++) {}
-                for (int i = 0; i < 10; i++) {
-                  /*foo*/
-                }
-                // violation below ''for' construct must use '{}'s.'
-                for (int b : abc);
-                for (int b : abc) {}
-                for (int b : abc) {
-                  /*foo*/
-                }
-              }
-            };
+        new InnerEmptyBlocks() {
+          boolean flag = true;
+          int[] abc = {
+            1, 2, 3, 4,
+          };
+
+          void foo() {
+            // violation below ''while' construct must use '{}'s.'
+            while (flag)
+              ;
+            while (flag) {}
+            while (flag) {
+              /*foo*/
+            }
+            // violation below ''do' construct must use '{}'s.'
+            do
+              ;
+            while (flag);
+            do {} while (flag);
+            do {
+              /*foo*/
+            } while (flag);
+            // violation below ''if' construct must use '{}'s.'
+            if (flag)
+              ;
+            if (flag) {
+              /*foo*/
+            }
+            if (flag) {
+              /*foo*/
+            }
+            // violation 2 lines below ''if' construct must use '{}'s.'
+            // violation 3 lines below ''else' construct must use '{}'s.'
+            if (flag)
+              ;
+            else
+              ;
+            // violation below 'Empty if block.'
+            if (flag) {
+            } else {
+            }
+            // violation 2 lines above 'Empty else block.'
+            if (flag) {
+              /*foo*/
+            } else {
+              /*foo*/
+            }
+            // violation below ''for' construct must use '{}'s.'
+            for (int i = 0; i < 10; i++)
+              ;
+            for (int i = 0; i < 10; i++) {}
+            for (int i = 0; i < 10; i++) {
+              /*foo*/
+            }
+            // violation below ''for' construct must use '{}'s.'
+            for (int b : abc)
+              ;
+            for (int b : abc) {}
+            for (int b : abc) {
+              /*foo*/
+            }
+          }
+        };
   }
 }
```

Zopsss commented:
I realized that the OG and formatted files were almost identical, expect formatter changed `;`'s location in some places and anonymous class' indentation other than this the formatted file and og file were same. So I just formatted the og file and removed inputformatted file. Some violations' location were changed so had to fix their messages to make test pass. And excluded this folder from google-java-format.sh

> Check PR Description / Analyze (pull_request) 

this failing CI can be resolved by restarting it. I have fixed the issue number in PR description

romani reviewed (approved):
Ok to merge
